server/portal/apps/site_search/api/views.py

```diff
@@ -0,0 +1,92 @@
+from django.http import JsonResponse
+from elasticsearch_dsl import Search
+from portal.libs.agave.operations import search as search_operation
+from portal.views.base import BaseApiView
+from django.conf import settings
+import logging
+logger = logging.getLogger(__name__)
+
+
+def cms_search(query_string, offset=0, limit=10):
+    cms_index = settings.ES_INDEX_PREFIX.format('cms')
+    cms_search = Search(index=cms_index)\
+        .query(
+        "query_string",
+        query=query_string,
+        default_operator="and",
+        fields=['title', 'body'])\
+        .highlight(
+        'body',
+        fragment_size=100)\
+        .highlight('title')\
+        .highlight_options(
+        pre_tags=["<b>"],
+        post_tags=["</b>"],
+        require_field_match=False)\
+        .extra(from_=offset, size=limit)
+
+    cms_search = cms_search.execute()
+    res = cms_search.hits
+    total = cms_search.hits.total.value
+
+    results = list(map(lambda x: {**x.to_dict(),
+                                  'highlight': x.meta.highlight.to_dict()},
+                       res))
+    return total, results
+
+
+def files_search(query_string, system, offset=0, limit=10):
+    res = search_operation(None, system, '/', offset=offset,
+                           limit=limit, query_string=query_string)
+    return (res['count'], res['listing'])
+
+
+class IndexView(BaseApiView):
+    """
+    Main workbench view.
+    """
+    template_name = 'portal/apps/workbench/index.html'
+
+    def dispatch(self, request, *args, **kwargs):
+        logger.debug(request.GET)
+        qs = request.GET.get('query_string', '')
+        page = request.GET.get('page', 1)
+        limit = 10
+        offset = (int(page) - 1) * limit
+        cms_total, cms_results = cms_search(qs, offset, limit)
+
+        response = {
+            'cms': {'count': cms_total,
+                    'listing': cms_results,
+                    'type': 'cms',
+                    'include': True}}
+
+        try:
+            public_conf = next(conf for conf
+                               in settings.PORTAL_DATAFILES_STORAGE_SYSTEMS
+                               if conf['scheme'] == 'public')
+            (public_total, public_results) = \
+                files_search(qs, public_conf['system'], offset, limit)
+            response['public'] = {'count': public_total,
+                                  'listing': public_results,
+                                  'type': 'file',
+                                  'include': True}
+        except StopIteration:
+            pass
```

Comment:
This could be refactored to remove the `try/except StopIteration`

```suggestion
        for public_conf in (conf for conf in settings.PORTAL_DATAFILES_STORAGE_SYSTEMS if conf['scheme'] == 'public'):
            (public_total, public_results) = \
                files_search(qs, public_conf['system'], offset, limit)
            response['public'] = {'count': public_total,
                                  'listing': public_results,
                                  'type': 'file',
                                  'include': True}
```

Reply:
There's a possibility that there will be multiple systems with a `public`/`community` scheme for some portals (apparently ptdatax does this?) I'd like to leave this for now and revisit the system lookup after break.

Reply:
My suggested change doesn't change the behavior (still handles multiple systems) but is to avoid the use of `next`/`StopIteration`

Reply:
I'd rather keep it as is for now since there's a possibility of there being multiple public/community systems (apparently ptdatax does this?) In that case the behavior *does* change since the original code will return results for the first system and your suggestion will return the last system. We'll need to revisit this logic after break.

Reply:
gotcha 👍